src/Stratis.Bitcoin/Consensus/Validators/BlockValidator.cs

```diff
@@ -100,15 +96,14 @@ public IntegrityValidator(IConsensusRuleEngine consensusRules, ILoggerFactory lo
         }
 
         /// <inheritdoc />
-        public void VerifyBlockIntegrity(Block block, ChainedHeader chainedHeader)
+        public ValidationContext VerifyBlockIntegrity(ChainedHeaderBlock chainedHeaderBlock)
         {
-            this.logger.LogTrace("({0}:'{1}')", nameof(chainedHeader), chainedHeader);
-
-            var validationContext = new ValidationContext { BlockToValidate = block, ChainedHeaderToValidate = chainedHeader };
+            this.logger.LogTrace("({0}:'{1}')", nameof(chainedHeaderBlock), chainedHeaderBlock);
 
-            this.consensusRules.IntegrityValidation(validationContext);
+            ValidationContext result = this.consensusRules.IntegrityValidation(chainedHeaderBlock);
 
             this.logger.LogTrace("(-)");
+            return result;
```

Comment:
Not only here, but if you add return value, you should consider changing exit logs

Reply:
fixed